### PR TITLE
Move bft block header creation/validation to common

### PR DESCRIPTION
**Pull request by rain-on**
Signed-off-by: Trent Mohay <trent.mohay@consensys.net>

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: consensus/ibft/src/integration-test/java/org/hyperledger/besu/consensus/ibft/support/TestContextBuilder.java [hunk 1]
```diff
@@ -273,28 +273,28 @@ private static ControllerAndState createControllerAndFinalState(
     final StubGenesisConfigOptions genesisConfigOptions = new StubGenesisConfigOptions();
     genesisConfigOptions.byzantiumBlock(0);
 
-    final ProtocolSchedule protocolSchedule = IbftProtocolSchedule.create(genesisConfigOptions);
+    final ProtocolSchedule protocolSchedule = BftProtocolSchedule.create(genesisConfigOptions);
 
     /////////////////////////////////////////////////////////////////////////////////////
     // From here down is BASICALLY taken from IbftBesuController
```
Comment (usmansaleem) [inline consensus/ibft/src/integration-test/java/org/hyperledger/besu/consensus/ibft/support/TestContextBuilder.java:279]: nit: doco reference need to be updated.

**Review by usmansaleem — APPROVED** [review]
LGTM

File: consensus/ibft/src/integration-test/java/org/hyperledger/besu/consensus/ibft/support/TestContextBuilder.java (hunk 1 above)
Comment (rain-on) [reply consensus/ibft/src/integration-test/java/org/hyperledger/besu/consensus/ibft/support/TestContextBuilder.java:279]: this is still valid - this class represents the IBFT-BesuController - i.e. this class is still specific to the consensus mechanism.

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftProtocolSchedule.java [hunk 2]
```diff
@@ -59,33 +59,33 @@ public static ProtocolSchedule create(final GenesisConfigOptions config) {
     return create(config, PrivacyParameters.DEFAULT, false);
   }
 
-  private static ProtocolSpecBuilder applyIbftChanges(
-      final IbftConfigOptions ibftConfig, final ProtocolSpecBuilder builder) {
+  private static ProtocolSpecBuilder applyBftChanges(
+      final IbftConfigOptions configOptions, final ProtocolSpecBuilder builder) {
```
[the remark this answers is not in the document]
Comment (rain-on) [reply consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftProtocolSchedule.java:63]: done

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftHelpers.java [hunk 3]
```diff
@@ -25,9 +22,8 @@
 import org.hyperledger.besu.ethereum.core.Util;
 
 import java.util.Collection;
-import java.util.Optional;
 
-public class IbftHelpers {
+public class BftHelpers {
 
   public static final Hash EXPECTED_MIX_HASH =
```
[the remark this answers is not in the document]
Comment (rain-on) [reply consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftHelpers.java:28]: I suspect they will be different - but happy to deal with that in a different PR as its _quite_ a challenging problem

**Pull request merged**